### PR TITLE
Added usage of __slots__ to CIM objects and CIM types

### PR DESCRIPTION
**Pull request by andy-maier**
**Note: This PR is on top of PR #2501 - review only the delta.**

For seeing the resource savings by this change, compare the output of `make resourcetest` between this PR and PR #2501. Results have been posted to issue #2509.

For details, see commit message.

### CODE REVIEW

File: docs/changes.rst [hunk 1]
```diff
@@ -37,6 +37,9 @@ Released: not yet
   ``pywbem.configure_logging()`` function that disables logging.
   (part of issue #86)
 
+* Reduced memory consumption of CIM objects and CIM types by using slots.
```
Comment (KSchopmeyer) [inline docs/changes.rst:40]: how about by replacing existing attributes with python slots.

**Comment by KSchopmeyer** [comment]
**QUESTION:** What does this do to mocker cache compatiblity with any existing cached names? If the user already has cached mocks and upgrades to this version of pywbem, I would assume that the existing pickles are obsolete. Does this mean that they wll get an error?  Or will it simply recreate the cache????

**Review by KSchopmeyer — COMMENTED** [review]
Made one minor comment and added a question about this change and mock caching.

File: docs/changes.rst (hunk 1 above)
Comment (andy-maier) [reply docs/changes.rst:40]: They are still attributes. I changed it to "... by defining their attributes to use Python slots."

**Comment by andy-maier** [comment]
Your question is a very good thought. I verified it, and alas, using a mock cache that was created before this change, with this change, results in:
```
  File "c:\u\andi\projects\pywbemtools\pywbemtools\pywbemcli\_pywbemcli_operations.py", line 642, in _load_mockenv
    mockenv = pickle.load(fp)
AttributeError: 'CIMQualifier' object has no attribute '__dict__'
```

File: pywbem/_cim_types.py [hunk 2]
```diff
@@ -179,28 +179,58 @@ class SlottedPickleMixin(object):
     default, without requiring that they define methods __getstate__() and
     __setstate__().
 
-    This mixin class provides these methods when running on Python 2. The
-    methods support classes with both slotted and non-slotted attributes.
+    That lead to the idea to provide these methods only when running on
```
Comment (KSchopmeyer) [inline pywbem/_cim_types.py:182]: Love english :-).  Change lead to led.

**Review by KSchopmeyer — COMMENTED** [review]
One small comment.

File: pywbem/_cim_types.py (hunk 2 above)
Comment (andy-maier) [reply pywbem/_cim_types.py]: DONE :-)

**Pull request closed without merging**